lusid-sdk-python/tests/test_instrument_master.py

```diff
@@ -0,0 +1,230 @@
+import unittest
+import requests
+import json
+import uuid
+import os
+from datetime import datetime
+import pytz
+import lusid
+import lusid.models as models
+from unittest import TestCase
+from collections import namedtuple
+from msrest.authentication import BasicTokenAuthentication
+from time import sleep
+
+try:
+    # Python 3.x
+    from urllib.request import pathname2url
+except ImportError:
+    # Python 2.7
+    from urllib import pathname2url
+
+class TestFinbourneApi(TestCase):
+    client = None
+    instrumentIds = []
+
+
+    @classmethod
+    def setUpClass(cls):
+
+        cls.ISIN_PROPERTY_KEY = "Instrument/default/Isin"
+        cls.SEDOL_PROPERTY_KEY = "Instrument/default/Sedol"
+        cls.FIGI_SCHEME = "Figi"
+        cls.CUSTOM_INTERNAL_SCHEME = "ClientInternal"
+
+        #   load configuration
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        with open(os.path.join(dir_path, "secrets.json"), "r") as secrets:
+            config = json.load(secrets)
+
+        token_url = os.getenv("FBN_TOKEN_URL", config["api"]["tokenUrl"])
+        username = os.getenv("FBN_USERNAME", config["api"]["username"])
+        password = pathname2url(os.getenv("FBN_PASSWORD", config["api"]["password"]))
+        client_id = pathname2url(os.getenv("FBN_CLIENT_ID", config["api"]["clientId"]))
+        client_secret = pathname2url(os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"]))
+        cls.api_url = os.getenv("FBN_LUSID_API_URL", config["api"]["apiUrl"])
+
+        token_request_body = ("grant_type=password&username={0}".format(username) +
+                              "&password={0}&scope=openid client groups".format(password) +
+                              "&client_id={0}&client_secret={1}".format(client_id, client_secret))
+
+        headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
+        okta_response = requests.post(token_url, data=token_request_body, headers=headers)
+
+        assert okta_response.status_code == 200
+
+        #   set the okta api token
+        cls.api_token = {"access_token": okta_response.json()["access_token"]}
+
+        credentials = BasicTokenAuthentication(TestFinbourneApi.api_token)
+        cls.client = lusid.LUSIDAPI(credentials, TestFinbourneApi.api_url)
+
+        response = cls.seed_instruments()
+
+    @classmethod
+    def tearDownClass(cls):
+        inst_list = ["BBG000C6K6G9",
+                     "BBG000C04D57",
+                     "BBG000FV67Q4",
+                     "BBG000BF0KW3",
+                     "BBG000BF4KL1",
+                     "BBG000BF6B57"]
+
+        for i in inst_list:
+            response = cls.client.delete_instrument(cls.FIGI_SCHEME, i)
+        # fbn_figi = cls.client.get_instrument("Figi", "BBG000C6K6G9")
+        print("JKH")
+    @classmethod
+    def seed_instruments(cls):
+
+        inst_isin = models.InstrumentProperty(cls.ISIN_PROPERTY_KEY, models.PropertyValue("GB00BH4HKS39"))
```

Comment:
it might aid readability to move these inline into the `properties=[]`, otherwise rename to `inst1_isin`, `inst2_isin` etc. rather than reusing to aid readability

Reply:
I changed the variable names to be clearer. I had a go at embedding but couldn't make it look clearer